ios/Framework/Core/Auth/LoginScreenlet/LoginScreenlet.swift

```diff
@@ -156,56 +189,34 @@ open class LoginScreenlet: BaseScreenlet, BasicAuthBasedType {
 
 	// MARK: Private methods
 
-	fileprivate func createLoginBasicInteractor() -> LoginBasicInteractor {
-		let interactor = LoginBasicInteractor(loginScreenlet: self)
-
-		interactor.onSuccess = {
-			self.loginDelegate?.screenlet?(self,
-					onLoginResponseUserAttributes: interactor.resultUserAttributes!)
-
-			if let ctx = SessionContext.currentContext, self.saveCredentials {
-				if ctx.storeCredentials() {
-					self.loginDelegate?.screenlet?(self,
-						onCredentialsSavedUserAttributes: interactor.resultUserAttributes!)
-				}
-			}
-		}
-
-		interactor.onFailure = {
-			self.loginDelegate?.screenlet?(self, onLoginError: $0)
-		}
-
-		return interactor
+	fileprivate func createLoginBasicInteractor() -> Interactor & LoginResult {
```

Comment:
I'd say this function should return `LoginBasicInteractor` instead of a generic type